docs/react-components-structure.md

```diff
@@ -0,0 +1,161 @@
+our first templates/pages components. Indeed, since there was no distinction
+between UI and business logic, templates were already full grown pages.
+
+So we just considered that the difference didn't stand in our case and got rid
+of `pages` folder and move the components to the `templates` folder.
+
+```
+src/
+├── ...
+└── web/
+    └── theme/
+        ├── atoms/
+        ├── molecules/
+        ├── organisms/
+        └── templates/
+```
+
+## Step 3: Split business and UI logic
+
+The core product continued to grow, we had more and more components and the sun
+was shiny. But after a while we felt that the theme was inconsistent. Even
+though it was the whole point of Atomic Design, we failed at building a
+homogeneous theme.
+
+So we stopped and looked back for a bit. What was wrong here?
+
+The root cause was that it was really hard to find existing components. It
+leaded to many dupplicates. The theme components were actually drowned in the
+middle of the business logic components.
+
+That's why we brought back the `pages` components back. They would be the React
+Components containing specific business logic, fetching data from the outside
+and giving life to UI components. In order to emphasize this decision, we
+created a new folder called `components` which would contain all the UI
+components (`atoms`, `molecules`, `organisms` and `templates`).
+
+```
+src/
+├── ...
+└── web/
+    └── theme/
+        ├── components/
+        |   ├── atoms/
+        |   ├── molecules/
+        |   ├── organisms/
+        |   └── templates/
+        └── pages/
+```
+
+And everything seemed pretty perfect here. The theme took back its consistency
+and it was easy to create new pages only by using the existing building blocks.
+
+## Step 4: Split pages and modules
+
+However, there was still one last thing that felt off. In `pages`, there would
+be components that would refer to only a small part of the page. It wasn't that
+annoying, but it made things fuzzier for external users. How would they know
+which component was the root page component and which one would only be a small
+module?
+
+That's why we decided to split the `pages` folder in two:
+
+* `pages` for the root components
+* `modules` for the smaller components that still contained business logic
+
+Additionnally, since everything is a component in React, we renamed the
+`components` folder into `ui`.
+
+```
+src/
+├── ...
+└── web/
+    └── theme/
+        ├── ui/
+        |   ├── atoms/
+        |   ├── molecules/
+        |   ├── organisms/
+        |   └── templates/
+        ├── modules/
+        └── pages/
+```
+
+## What's next?
+
+We are now pretty confident that this structure is stable enough and will help
+you through the creation of brand new themes. We have already built oursleves
+two custom themes for production stores based on this structure, and it was a
+breeze.
+
+Additionally, if things were to change in the future, rest assured that we will
```

Comment:
You know I love bold text… hence the proposal to boldify the reassurance part :smiley: